### PR TITLE
Remove Axios module & added API functionality

### PR DESCRIPTION
**Pull request by xvoorvaa**
Code is not tested yet. Please do not merge it yet

This should fix issue #79

### CODE REVIEW

**Comment by xvoorvaa** [comment]
@wslyvh `/server` has nothing to do with the NextJS app, correct?

**Comment by wslyvh** [comment]
I think it does. Next uses the code in server to build/import all the data? cc @pabloVoorvaart 

But I'd recommend to remove it either way. 
Using cross-fetch for example makes it easier to share the same codebase on app/server

**Comment by xvoorvaa** [comment]
@greatsamist Wanna confirm if you are also working on this. If not, good :). if so, then I just wasted my time :(.

**Comment by xvoorvaa** [comment]
`EventEntry.tsx` & `OrganizationEntry.tsx` are tested

**Comment by greatsamist** [comment]
> confirm
No, I am not. great job

**Comment by pabloVoorvaart** [comment]
can this be merged?

**Comment by xvoorvaa** [comment]
I get a strange `500` status code when adding an organization, but it does create it.

File: app/api/organizations/route.ts [hunk 1]
```diff
@@ -17,30 +44,35 @@ export async function GET() {
...
-      const folderName = `data/organizations`
-      const fileName = `${body.name.replace(/ /g, '_').toLowerCase()}.json`
-      await AddOrUpdateFile(fileName, JSON.stringify(body), folderName)
-    }
-    return NextResponse.json(body)
-  } catch (e) {
-    console.log(e)
-    return NextResponse.json({ error: 'Malformed request' }, { status: 400 })
-  }
+export function POST(request: NextRequest) {
```
Comment (xvoorvaa) [inline app/api/organizations/route.ts:47]: Hmm maybe a bit too cursed...

File: app/api/organizations/route.ts [hunk 2]
```diff
@@ -2,6 +2,33 @@ import { NextRequest, NextResponse } from 'next/server'
...
 import Session from '@/utils/session'
+import EventController from '@/server/controller/event'
+import { IEvent } from '@/server/model/event'
+import { IOrganization } from '@/server/model/organization'
+import { generateId } from '@/server/utils'
+
+/*
+ * The events of the organization will also be deleted
+ */
+export async function DELETE(req: NextRequest): Promise<NextResponse> {
```
Comment (xvoorvaa) [inline app/api/organizations/route.ts:13]: Need to check Session

File: server/controller/organization.ts [hunk 3]
```diff
@@ -21,6 +21,11 @@ export default class OrganizationController {
     return org
   }
 
+  public async editOrganization(organization: IOrganization): Promise<Organization> {
```
Comment (xvoorvaa) [inline server/controller/organization.ts:24]: Not sure if this is ok

**Comment by xvoorvaa** [comment]
I've successfully transitioned from using Axios to the native Fetch API for all our HTTP requests. I've managed to update all the functions to work correctly. I've tested on each of them to confirm their functionality.

I've also added a few comments on this PR to with questions or additional notes. I'd appreciate your thoughts on these.

@wslyvh @pabloVoorvaart, let me know what you think!

File: app/api/organizations/route.ts (hunk 2 above)
Comment (wslyvh) [reply app/api/organizations/route.ts:13]: The same is probably true for any method with POST, PUT or DELETE

**Comment by wslyvh** [comment]
I think it looks good. Left some comments on authorization using SIWE

**Comment by xvoorvaa** [comment]
Hey @greatsamist, I'm having trouble understanding why `POST` and `PATCH` requests aren't sending cookies to the API. Have you encountered this issue before?

I found a [Stackoverflow discussion](https://stackoverflow.com/questions/76274546/next-js-does-not-send-cookies-with-fetch-request-even-though-credentials-are-inc) that offers a potential solution, but I'm not sure about its security implications, and if this is the correct way to do on NextJS.

I've tried using `credentials: include`, but it hasn't resolved the problem. Any suggestions?
You can check out my last [commit](https://github.com/streamethorg/streameth-platform/pull/91/commits/f5c9a2888eff226daa2fb18a33f6c6d0e8cdd09f) for more info.

**Comment by greatsamist** [comment]
Using `credentials: include` is currently ineffective in next 13
This will also be my first time with this issue, have you tried using the cookie fn to get cookies [next13 cookies](https://nextjs.org/docs/app/api-reference/functions/cookies#good-to-know) then add them to the headers [Stackoverflow usage](https://stackoverflow.com/a/76280543)

**Comment by xvoorvaa** [comment]
Hey y'all, everything is tested and the session is being checked, before the action actually happens.

@wslyvh & @pabloVoorvaart could you give me a final review please

File: server/controller/event.ts [hunk 4]
```diff
@@ -25,6 +25,11 @@ export default class EventController {
     return evt
   }
 
+  public async editEvent(event: IEvent, organizationId: IEvent['organizationId']): Promise<void> {
+    this.deleteEvent(event.id, organizationId)
```
Comment (greatsamist) [inline server/controller/event.ts:29]: This will delete the particular event completely before calling `this.create` Please double check

**Review by greatsamist — CHANGES_REQUESTED** [review]

**Comment by wslyvh** [comment]
Seems like this has become a bigger change than anticipated. But nice refactoring! Looks good overall.
Good catch by @greatsamist on deletion.

On a more general note, I see a lot of then/catch promises being introduced here.
While there's nothing wrong, we've typically used an await/try/catch pattern.
For consistency, I would prefer one way or the other. Ideally the latter, since that's been the common approach so far.

File: server/controller/event.ts (hunk 4 above)
Comment (xvoorvaa) [reply server/controller/event.ts]: editOrganization & editEvent works as intended now

**Review by greatsamist — APPROVED** [review]

**Comment by xvoorvaa** [comment]
So I was aware about 2, but it has to do with the front-end. I think it is best to fix that in your refreshed Admin UI.

**Pull request merged**